app/controllers/incomeSources/add/AddBusinessNameController.scala

```diff
@@ -70,13 +70,10 @@ class AddBusinessNameController @Inject()(authenticate: AuthenticationPredicate,
 
   private lazy val journeyType: JourneyType = JourneyType(Add, SelfEmployment)
 
-  private def getBusinessName(isChange: Boolean)
+  private def getBusinessName(isChange: Boolean, sessionData: Option[UIJourneySessionData])
                              (implicit user: MtdItUser[_]): Future[Option[String]] = {
     if (isChange)
-      sessionService.getMongoKeyTyped[String](AddIncomeSourceData.businessNameField, journeyType).flatMap {
-        case Right(nameOpt) => Future.successful(nameOpt)
-        case Left(ex) => Future.failed(ex)
-      }
+      Future.successful(sessionData.get.addIncomeSourceData.get.businessName)
```

Comment:
Again, there is no need to create a future here. The sessionData is already available.